azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/NetworkSecurityRuleApiLiveTest.java

```diff
@@ -0,0 +1,214 @@
+            }
+         }, 60 * 2 * 1000 /* 2 minute timeout */).apply(uri);
+      }
+
+      super.tearDown();
+   }
+
+   @Test(groups = "live")
+   public void deleteNetworkSecurityRuleDoesNotExist() {
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      URI uri = ruleApi.delete(UNKNOWN_RULE_NAME);
+      assertNull(uri);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "deleteNetworkSecurityRuleDoesNotExist")
+   public void createNetworkSecurityRule() {
+      final NetworkSecurityRule rule = createRule();
+      assertNotNull(rule);
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      NetworkSecurityRule result = ruleApi.createOrUpdate(rule.name(), rule.properties());
+      assertNotNull(result);
+      assertEquals(result.name(), rule.name());
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void getNetworkSecurityRule() {
+      final NetworkSecurityRule rule = createRule();
+      assertNotNull(rule);
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      NetworkSecurityRule result = ruleApi.get(rule.name());
+      assertNotNull(result);
+      assertNotNull(result.etag());
+      assertEquals(result.name(), rule.name());
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void getNetworkSecurityDefaultRule() {
+      String defaultRuleName = "AllowVnetInBound";
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      NetworkSecurityRule result = ruleApi.getDefaultRule(defaultRuleName);
+
+      assertNotNull(result);
+      assertNotNull(result.etag());
+      assertEquals(result.name(), defaultRuleName);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void listNetworkSecurityRules() {
+      final NetworkSecurityRule rule = createRule();
+      assertNotNull(rule);
+
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      List<NetworkSecurityRule> result = ruleApi.list();
+
+      assertNotNull(result);
+      assertEquals(result.size(), 2);
+
+      boolean rulePresent = false;
+      for (int i = 0; i < result.size(); i++) {
+         if (result.get(i).name().equals(rule.name())) {
+            rulePresent = true;
+            break;
+         }
+      }
+      assertTrue(rulePresent);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityRule")
+   public void listDefaultSecurityRules() {
+      final NetworkSecurityRuleApi ruleApi = api.getNetworkSecurityRuleApi(resourcegroup, DEFAULT_NSG_NAME);
+      List<NetworkSecurityRule> result = ruleApi.listDefaultRules();
+
+      assertNotNull(result);
+      assertTrue(result.size() > 0);
```

Comment:
Default rules are automatically created as part of the resource group.  It is not well defined what the default rules are and whether or not that may change over time.  These aren't created by code in azurecompute-arm, but rather are generated as part of creating the network security group.

So, suggestions for a test other than size?